include/tscore/Diags.h

```diff
@@ -33,236 +33,9 @@
 
 #pragma once
 
-#include <cstdarg>
-#include "ink_mutex.h"
-#include "Regex.h"
-#include "ink_apidefs.h"
-#include "ContFlags.h"
-#include "ink_inet.h"
-#include "BaseLogFile.h"
+#include "DiagsTypes.h"
 #include "SourceLocation.h"
-
-#define DIAGS_MAGIC 0x12345678
-#define BYTES_IN_MB 1000000
-
-class Diags;
-
-// extern int diags_on_for_plugins;
-enum DiagsTagType {
```

Comment:
Note that for this file all the removed code is the type declarations which simply got moved into the new DiagsTypes.h. This avoids a circular include dependency by allowing LogMessage to include DiagsTypes.h without Diags.h, the latter of which also needs to include LogMessage.h. (I also think it conceptually makes sense to separate them anyway.)